src/OmnichannelChatSDK.ts

```diff
@@ -212,81 +213,109 @@ class OmnichannelChatSDK {
     }
 
-    private async retryLoadAMSClient() : Promise<AmsClient> {
+    private async retryLoadAMSClient(): Promise<AmsClient> {
         // Constants for retry logic
         const RETRY_DELAY_MS = 1000;
-        const MAX_RETRY_COUNT = 5;
+        const MAX_RETRY_COUNT = 30;
         let retryCount = 0;
 
         while (retryCount < MAX_RETRY_COUNT) {
             if (this.AMSClient) {
                 return this.AMSClient;
             }
-
             await sleep(RETRY_DELAY_MS);
             retryCount++;
         }
+        console.error("Failed to load AMSClient after multiple attempts.");
         return null;
     }
 
-    private async getAMSClient() : Promise<AmsClient> {
+    private async getAMSClient(): Promise<AmsClient> {
+
+        //return null to do not break promisse creation
+        if (this.isAMSClientAllowed === false) {
+            return null;
+        }
 
         if (this.AMSClientLoadCurrentState === AMSClientLoadStates.NOT_LOADED && this.liveChatVersion === LiveChatVersion.V1) {
             return null;
         }
         switch (this.AMSClientLoadCurrentState) {
         case AMSClientLoadStates.LOADED:
+            this.debug && console.log("AMSClient is already loaded");
             return this.AMSClient;
         case AMSClientLoadStates.LOADING:
+            this.debug && console.log("AMSClient is loading, waiting for it to be ready");
             return await this.retryLoadAMSClient();
-
         case AMSClientLoadStates.ERROR:
         case AMSClientLoadStates.NOT_LOADED:
+            this.debug && console.log("AMSClient is not loaded, loading now");
             await this.loadAmsClient();
             return this.AMSClient;
-
         default:
             return null;
         }
     }
 
-    private async loadInitComponents() : Promise<void>{
+    private async loadInitComponents(): Promise<void> {
         this.scenarioMarker.startScenario(TelemetryEvent.InitializeComponents);
 
         const supportedLiveChatVersions = [LiveChatVersion.V1, LiveChatVersion.V2];
         if (!supportedLiveChatVersions.includes(this.liveChatVersion)) {
             exceptionThrowers.throwUnsupportedLiveChatVersionFailure(new Error(ChatSDKErrorName.UnsupportedLiveChatVersion), this.scenarioMarker, TelemetryEvent.InitializeComponents);
         }
-
-        this.isInitialized = true;
+        // we need this version validation, until we remove all v1 code, pending task.
+        if (this.liveChatVersion === LiveChatVersion.V2) {
+            this.ACSClient = new ACSClient(this.acsClientLogger);
+        } else {
+            this.IC3Client = await this.getIC3Client();
+        }
         this.scenarioMarker.completeScenario(TelemetryEvent.InitializeComponents);
     }
 
-    private async loadAmsClient() : Promise<void> {
-        this.scenarioMarker.startScenario(TelemetryEvent.InitializeMessagingClient);
-        try {
+    private evaluateAMSAvailability(): boolean {
```

Comment:
`loadAMSIfAttachmentEnabled()`?

Reply:
I feel the current name is more descriptive, since is not really loading , it just check if it should  ....